Source/Diagnostics/WarpXIO.cpp

```diff
@@ -492,6 +492,18 @@ WarpX::WritePlotFile () const
         output_geom = Geom();
     }
 
+#ifdef WARPX_USE_OPENPMD
+    if (dump_openpmd){
+        // Write openPMD format: only for level 0
+        std::string filename = amrex::Concatenate("diags/hdf5/data", istep[0]);
+        filename += ".h5";
```

Comment:
you could make the file ending a WarpX option which defaults to `".h5"` in your case.

that way, a user is able to load another dynamic `libopenPMD.[so|dylib]` library with, e.g. ADIOS support, and can switch the whole I/O backend. Unsupported I/O backends will throw in the constructor of `Series`, which you might want to catch and handle gracefully (could also happen if HDF5 backend is not found).